### PR TITLE
[SDL-0296] OnSystemCapabilityUpdated scripts

### PR DESCRIPTION
**Pull request by aderiabin**
This PR is **[ready]** for review.

### Summary
Check OnSystemCapabilityUpdated processing 

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/UpdateVideoStreamingCapabilities/002_10_OnSystemCapabilityUpdated_not_existing_appid.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
+--[[ Local Variables ]]
+local appSessionId = 1
+local isSubscribe = true
+
+local vsc = common.cloneTable(common.anotherVideoStreamingCapabilityWithOutAddVSC)
+
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_10_OnSystemCapabilityUpdated_not_existing_appid.lua:12]: Please all `--[[ Local Functions ]]`

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua [hunk 2]
```diff
@@ -0,0 +1,107 @@
+---------------------------------------------------------------------------------------------------
+-- https://adc.luxoft.com/jira/browse/FORDTCN-7008
+
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
+--[[ Local Variables ]]
+
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua:9]: remove empty line

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  sendOnSystemCapabilityUpdatedMultipleApps, { appSessionId1, notExpected, notExpected })
+common.Step("OnSystemCapabilityUpdated appID = 2",
+  sendOnSystemCapabilityUpdatedMultipleApps, { appSessionId2, notExpected, notExpected })
+common.Step("OnSystemCapabilityUpdated no appID",
+  sendOnSystemCapabilityUpdatedMultipleApps, { appSessionIdAbsent, notExpected, notExpected })
+
+common.Title("Postconditions")
+common.Step("Stop SDL", common.postconditions)
+
+common.Title("Postconditions")
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua:106]: Please remove redundant postcondition

File: test_scripts/UpdateVideoStreamingCapabilities/002_2_OnSystemCapabilityUpdated_without_AddVSC.lua [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+-- [[ Required Shared libraries ]]
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
+--[[ Local Variables ]]
+local appSessionId = 1
+local expected = 1
+local isSubscribe = true
+
+local vsc = common.cloneTable(common.anotherVideoStreamingCapabilityWithOutAddVSC)
+vsc.additionalVideoStreamingCapabilities = nil
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_2_OnSystemCapabilityUpdated_without_AddVSC.lua:13]: `anotherVideoStreamingCapabilityWithOutAddVSC` parameter is defined as value wihout additional VSC

File: test_scripts/UpdateVideoStreamingCapabilities/002_4_OnSystemCapabilityUpdated_in_range_minsize_maxsize_array.lua [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  common.Title("Preconditions")
+  common.Step("Clean environment", common.preconditions)
+  common.Step("Set HMI Capabilities", common.setHMICapabilities)
+  common.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { common.hmiDefaultCapabilities })
+  common.Step("RAI", common.registerAppWOPTU)
+  common.Step("Activate App", common.activateApp)
+  common.Step("GetSystemCapability with subscribe = true", common.getSystemCapability, { isSubscribe })
+
+  common.Title("Test")
+  common.Step("OnSystemCapabilityUpdated in range " .. parameter .. " " .. value,
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_4_OnSystemCapabilityUpdated_in_range_minsize_maxsize_array.lua:27]: Propose to not execute all steps for different values from `arraySize` but only current with sending of the notification

File: test_scripts/UpdateVideoStreamingCapabilities/002_5_OnSystemCapabilityUpdated_out_of_range_minsize_maxsize_array.lua [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  common.Title("Preconditions")
+  common.Step("Clean environment", common.preconditions)
+  common.Step("Set HMI Capabilities", common.setHMICapabilities)
+  common.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { common.hmiDefaultCapabilities })
+  common.Step("RAI", common.registerAppWOPTU)
+  common.Step("Activate App", common.activateApp)
+  common.Step("GetSystemCapability with subscribe = true", common.getSystemCapability, { isSubscribe })
+
+  common.Title("Test")
+  common.Step("OnSystemCapabilityUpdated out of range " .. parameter .. " " .. value,
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_5_OnSystemCapabilityUpdated_out_of_range_minsize_maxsize_array.lua:27]: Propose to not execute all steps for different values from `arraySize` but only current with sending of the notification

File: test_scripts/UpdateVideoStreamingCapabilities/002_9_OnSystemCapabilityUpdated_missing_all_param.lua [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
+--[[ Local Variables ]]
+local appSessionId = 1
+local expected = 1
+local isSubscribe = true
+
+local vsc = {}
+
+
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/002_9_OnSystemCapabilityUpdated_missing_all_param.lua:14]: please remove redundant new line

File: test_scripts/UpdateVideoStreamingCapabilities/002_2_OnSystemCapabilityUpdated_without_AddVSC.lua (hunk 4 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_2_OnSystemCapabilityUpdated_without_AddVSC.lua:27]: @aderiabin Yes, you are right. But with this line of code we can be sure that it is.

File: test_scripts/UpdateVideoStreamingCapabilities/002_9_OnSystemCapabilityUpdated_missing_all_param.lua (hunk 7 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_9_OnSystemCapabilityUpdated_missing_all_param.lua]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_5_OnSystemCapabilityUpdated_out_of_range_minsize_maxsize_array.lua (hunk 6 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_5_OnSystemCapabilityUpdated_out_of_range_minsize_maxsize_array.lua:41]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_4_OnSystemCapabilityUpdated_in_range_minsize_maxsize_array.lua (hunk 5 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_4_OnSystemCapabilityUpdated_in_range_minsize_maxsize_array.lua:41]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua (hunk 3 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua (hunk 2 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_10_OnSystemCapabilityUpdated_not_existing_appid.lua (hunk 1 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/002_10_OnSystemCapabilityUpdated_not_existing_appid.lua:25]: Done (see 43db3b7)

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      systemCapabilityType = "VIDEO_STREAMING",
+      videoStreamingCapability = common.getVideoStreamingCapability()
+    }
+  }
+  local hmiParams = common.cloneTable(mobileParams)
+  if pAppId then
+    hmiParams.appID = common.getHMIAppId(pAppId)
+  end
+  common.getHMIConnection():SendNotification("BasicCommunication.OnSystemCapabilityUpdated", hmiParams)
+  common.getMobileSession(1):ExpectNotification("OnSystemCapabilityUpdated", mobileParams)
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua:59]:
```suggestion
  common.getMobileSession(appSessionId1):ExpectNotification("OnSystemCapabilityUpdated", mobileParams)
```

File: test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+  :Times(pTimesAppId1)
+  :ValidIf(function(_, data)
+    if pTimesAppId1 == 1 and (not common.isTableEqual(mobileParams, data.payload)) then
+      return false, "Parameters of the notification received by App1 are incorrect: \nExpected: "
+        .. common.toString(mobileParams) .. "\nActual: " .. common.toString(data.payload)
+    end
+    return true
+  end)
+  common.getMobileSession(2):ExpectNotification("OnSystemCapabilityUpdated", mobileParams)
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/002_11_OnSystemCapabilityUpdated_multiple_apps.lua:68]:
```suggestion
  common.getMobileSession(appSessionId2):ExpectNotification("OnSystemCapabilityUpdated", mobileParams)
```

File: test_scripts/UpdateVideoStreamingCapabilities/002_6_1_OnSystemCapabilityUpdated_invalid.lua [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  common.Step("Clean environment", common.preconditions)
+  common.Step("Set HMI Capabilities", common.setHMICapabilities)
+  common.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { common.hmiDefaultCapabilities })
+  common.Step("RAI", common.registerAppWOPTU)
+  common.Step("Activate App", common.activateApp)
+  common.Step("GetSystemCapability with subscribe = true", common.getSystemCapability, { isSubscribe })
+
+  common.Title("Test")
+  common.Step("OnSystemCapabilityUpdated " .. type, common.sendOnSystemCapabilityUpdated,
+    {appSessionId, notExpected, value })
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/002_6_1_OnSystemCapabilityUpdated_invalid.lua:76]:
```suggestion
    { appSessionId, notExpected, value })
```

**Pull request merged**